prism_compile.c

```diff
@@ -1877,49 +1980,13 @@ pm_compile_node(rb_iseq_t *iseq, const pm_node_t *node, LINK_ANCHOR *const ret,
       case PM_CALL_NODE: {
         pm_call_node_t *call_node = (pm_call_node_t *) node;
 
-        ID method_id = pm_constant_id_lookup(scope_node, call_node->name);
-        int flags = 0;
-        struct rb_callinfo_kwarg *kw_arg = NULL;
-
         if (call_node->receiver == NULL) {
             PM_PUTSELF;
         } else {
             PM_COMPILE_NOT_POPPED(call_node->receiver);
         }
 
-        int orig_argc = pm_setup_args(call_node->arguments, &flags, &kw_arg, iseq, ret, src, popped, scope_node, dummy_line_node, parser);
-
-        const rb_iseq_t *block_iseq = NULL;
-        if (call_node->block != NULL && PM_NODE_TYPE_P(call_node->block, PM_BLOCK_NODE)) {
-            // Scope associated with the block
-            pm_scope_node_t next_scope_node;
-            pm_scope_node_init(call_node->block, &next_scope_node, scope_node, parser);
-
-            block_iseq = NEW_CHILD_ISEQ(next_scope_node, make_name_for_block(iseq), ISEQ_TYPE_BLOCK, lineno);
-            ISEQ_COMPILE_DATA(iseq)->current_block = block_iseq;
-        }
-        else {
-            if (node->flags & PM_CALL_NODE_FLAGS_VARIABLE_CALL) {
-                flags |= VM_CALL_VCALL;
-            }
-
-            if (call_node->block != NULL) {
-                PM_COMPILE_NOT_POPPED(call_node->block);
-                flags |= VM_CALL_ARGS_BLOCKARG;
-            }
-
-            if (!flags) {
-                flags |= VM_CALL_ARGS_SIMPLE;
-            }
-        }
-
-        if (call_node->receiver == NULL) {
-            flags |= VM_CALL_FCALL;
-        }
-
-        ADD_SEND_R(ret, &dummy_line_node, method_id, INT2FIX(orig_argc), block_iseq, INT2FIX(flags), kw_arg);
-        PM_POP_IF_POPPED;
-        return;
+        pm_compile_call(iseq, call_node, ret, src, popped, scope_node);
```

Comment:
This looks like it will fall through, I'm assuming that's not intentional?

Reply:
No, it's a bug.  Eileen pushed a fix for it 😅